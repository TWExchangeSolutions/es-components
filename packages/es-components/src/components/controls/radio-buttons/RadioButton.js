import React from 'react';
import PropTypes from 'prop-types';
import styled, { ThemeProvider, withTheme } from 'styled-components';
import { noop } from 'lodash';
import viaTheme from 'es-components-via-theme';

import Label from '../Label';
import getRadioFillVariables from './radio-fill-variables';

function radioFill(color) {
  return `
    background-color: ${color};
    border-radius: 100%;
    content: '';
    display: block;
    height: 13px;
    left: 3px;
    position: absolute;
    width: 13px;
    top: 3px;
    transition: background 0.25s linear;
  `;
}

const RadioLabel = styled(Label)`
  color: ${props => (props.error ? props.theme.colors.danger : 'inherit')};
  display: ${props => (props.inline ? 'inline-flex' : 'inherit')};
  font-size: ${props => props.theme.sizes.baseFontSize}px;
  font-weight: normal;
  margin-right: ${props => (props.inline ? '20px' : 'initial')};
  min-height: 25px;
  position: relative;
  padding: 5px 0;
  text-transform: none;

  &:hover .radio-fill:before {
    ${props => radioFill(props.hoverFillColor)};
  }
`;

const RadioInput = styled.input`
  opacity: 0;
  position: absolute;

  &:focus ~ .radio-fill {
    box-shadow: 0 0 3px 3px ${props => props.theme.colors.inputFocus};
  }
`;

const RadioText = styled.span`
  align-self: center;
  margin-left: 30px;
`;

const RadioDisplay = styled.span`
  border: 3px solid ${props => props.borderColor};
  border-radius: 100%;
  box-sizing: border-box;
  display: block;
  height: 25px;
  position: absolute;
  width: 25px;

  &:before {
    ${props => radioFill(props.fill)};
  }
`;

export function RadioButton({
  optionText,
  name,
  checked = false,
  id,
  isDisabled = false,
  inline = true,
  onClick = noop,
  value,
  hasError = false,
  theme,
  ariaHide,
  ...radioProps
}) {
  const { hover, fill } = getRadioFillVariables(
    checked,
    isDisabled,
    hasError,
    theme.colors
  );
  const radioDisplayFill = checked ? fill : theme.colors.white;

  const labelProps = {
    checked,
    inline,
    disabled: isDisabled,
    htmlFor: id,
    hoverFillColor: hover,
    error: hasError
  };

  return (
    <ThemeProvider theme={theme}>
      <RadioLabel {...labelProps}>
        <RadioInput
          type="radio"
          name={name}
          id={id}
          onClick={onClick}
          value={value}
          disabled={isDisabled}
          {...radioProps}
        />
        <RadioDisplay
          className="radio-fill"
          borderColor={fill}
          fill={radioDisplayFill}
        />
        <RadioText aria-hidden={ariaHide}>{optionText}</RadioText>
      </RadioLabel>
    </ThemeProvider>
  );
}

RadioButton.propTypes = {
  optionText: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  checked: PropTypes.bool,
  id: PropTypes.string,
  isDisabled: PropTypes.bool,
  inline: PropTypes.bool,
  onClick: PropTypes.func,
  value: PropTypes.any,
  ariaHide: PropTypes.bool,
  hasError: PropTypes.bool,
  /**
   * Theme object used by the ThemeProvider,
   * automatically passed by any parent component using a ThemeProvider
   */
  theme: PropTypes.object
};

RadioButton.defaultProps = {
  theme: viaTheme
};

export default withTheme(RadioButton);
