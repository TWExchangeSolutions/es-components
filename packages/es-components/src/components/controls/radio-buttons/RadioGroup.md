Group radio buttons together using a `RadioGroup` component. Providing a `legendText` prop will render a legend with the grouped radio buttons. Individual options can also be disabled.


```
const options = [{
  optionText: 'Hiking',
  optionValue: 'hiking'
}, {
  optionText: 'Biking',
  optionValue: 'biking'
}, {
  optionText: 'Kayaking',
  optionValue: 'kayaking'
}, {
  optionText: 'Camping',
  optionValue: 'camping',
  isDisabled: true
}];

class RadioGroupExample extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      value: ''
    };

    this.optionChanged = this.optionChanged.bind(this);
  }

  optionChanged(event) {
    const value = event.target.value;
    this.setState({ value });
  }

  render() {
    return (
      <RadioGroup
        name="recreational-activities"
        legendText="Recreational activities"
        radioOptions={options}
        onClick={this.optionChanged}
        value={this.state.value}
      />
    );
  }
}

<RadioGroupExample />
```

Setting the `inline` option to false will stack the radio buttons

```
const options = [{
  optionText: 'Meat',
  optionValue: 'meat'
}, {
  optionText: 'Seafood',
  optionValue: 'seafood'
}, {
  optionText: 'Vegetarian',
  optionValue: 'vegetarian'
}];

class RadioGroupExample extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      value: ''
    };

    this.optionChanged = this.optionChanged.bind(this);
  }

  optionChanged(event) {
    this.setState({ value: event.target.value });
  }

  render() {
    return (
      <RadioGroup
        name="plates"
        radioOptions={options}
        inline={false}
        onClick={this.optionChanged}
        value={this.state.value}
      />
    );
  }
}

<RadioGroupExample />
```

Each radio is displayed as an error when the `hasError` prop is true. An errored radio group with a default checked option is rendered with a filled radio button.

```
const options = [{
  optionText: 'Planes',
  optionValue: 'plane'
}, {
  optionText: 'Trains',
  optionValue: 'train'
}, {
  optionText: 'Automobiles',
  optionValue: 'automobile'
}];

class RadioGroupExample extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      value: 'plane'
    };

    this.optionChanged = this.optionChanged.bind(this);
  }

  optionChanged(event) {
    this.setState({ value: event.target.value });
  }

  render() {
    return (
      <RadioGroup
        name="transports"
        radioOptions={options}
        onClick={this.optionChanged}
        value={this.state.value}
        hasError
      />
    );
  }
}

<RadioGroupExample />
```

Each radio is disabled when the `disableAllOptions` prop is true. A disabled radio group with a default checked option is rendered with a filled radio button.

```
const options = [{
  optionText: 'Apple',
  optionValue: 'apple'
}, {
  optionText: 'Banana',
  optionValue: 'banana'
}, {
  optionText: 'Pear',
  optionValue: 'pear'
}];

class RadioGroupExample extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      value: 'pear'
    };

    this.optionChanged = this.optionChanged.bind(this);
  }

  optionChanged(event) {
    this.setState({ value: event.target.value });
  }

  render() {
    const extra = (<div>Some Descriptive Content about fruit</div>);
    return (
      <RadioGroup
        name="fruits"
        legendContent="Fruits"
        radioOptions={options}
        value={this.state.value}
        disableAllOptions
        extraContent={extra}
      />
    );
  }
}

<RadioGroupExample />
```
