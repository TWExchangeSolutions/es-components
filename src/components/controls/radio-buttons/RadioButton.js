import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { noop } from 'lodash';

import Label from '../Label';
import { colors, sizes } from '../../theme';

import getRadioFillVariables from './radio-fill-variables';

function radioFill(color) {
  return `
    background-color: ${color};
    border-radius: 100%;
    content: '';
    display: block;
    height: 13px;
    left: 3px;
    position: absolute;
    width: 13px;
    top: 3px;
    transition: background 0.25s linear;
  `;
}

const RadioLabel = styled(Label)`
  color: ${props => (props.error ? colors.danger : 'inherit')};
  display: ${props => (props.inline ? 'inline-flex' : 'inherit')};
  font-size: ${sizes.baseFontSize}px;
  font-weight: normal;
  margin-right: ${props => (props.inline ? '20px' : 'initial')};
  min-height: 25px;
  position: relative;
  padding: 5px 0;
  text-transform: none;

  &:hover .radio-fill:before { ${props => radioFill(props.hoverFillColor)} }
`;

const RadioInput = styled.input`
  opacity: 0;
  position: absolute;

  &:focus ~ .radio-fill {
    box-shadow: 0 0 3px 3px #83bffc
  }
`;

const RadioText = styled.span`
  align-self: center;
  margin-left: 30px;
`;

const RadioDisplay = styled.span`
  border: 3px solid ${props => props.borderColor};
  border-radius: 100%;
  box-sizing: border-box;
  display: block;
  height: 25px;
  position: absolute;
  width: 25px;

  &:before { ${props => radioFill(props.fill)} }
`;

function RadioButton({
  optionText,
  name,
  checked = false,
  id,
  disabled = false,
  inline = true,
  onChange = noop,
  value,
  isInErrorState = false,
  ...radioProps
}) {
  const { hover, fill } = getRadioFillVariables(
    checked,
    disabled,
    isInErrorState
  );
  const radioDisplayFill = checked ? fill : colors.white;

  const labelProps = {
    checked,
    inline,
    disabled,
    htmlFor: id,
    hoverFillColor: hover,
    error: isInErrorState
  };

  return (
    <RadioLabel {...labelProps}>
      <RadioInput
        type="radio"
        name={name}
        id={id}
        onChange={onChange}
        value={value}
        disabled={disabled}
        {...radioProps}
      />
      <RadioDisplay
        className="radio-fill"
        borderColor={fill}
        fill={radioDisplayFill}
      />
      <RadioText>{optionText}</RadioText>
    </RadioLabel>
  );
}

RadioButton.propTypes = {
  optionText: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  checked: PropTypes.bool,
  id: PropTypes.string,
  disabled: PropTypes.bool,
  inline: PropTypes.bool,
  onChange: PropTypes.func,
  value: PropTypes.any,
  isInErrorState: PropTypes.bool
};

export default RadioButton;
