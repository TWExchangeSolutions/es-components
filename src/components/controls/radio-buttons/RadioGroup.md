Group radio buttons together using a `RadioGroup` component.

```
const options = [{
  optionText: 'Red',
  optionValue: 'red'
}, {
  optionText: 'Green',
  optionValue: 'green'
}, {
  optionText: 'Blue',
  optionValue: 'blue'
}];

<RadioGroup name="colors" radioOptions={options} checkedValue="green" />
```

Providing a `legendText` option will render a legend with the grouped radio buttons.

```
const options = [{
  optionText: 'Hiking',
  optionValue: 'hiking'
}, {
  optionText: 'Biking',
  optionValue: 'biking'
}, {
  optionText: 'kayaking',
  optionValue: 'kayaking'
}];

<RadioGroup
  name="recreational-activities"
  legendText="Recreational activities"
  radioOptions={options}
/>
```

Setting the `inline` option to false will stack the radio buttons

```
const options = [{
  optionText: 'Meat',
  optionValue: 'meat'
}, {
  optionText: 'Seafood',
  optionValue: 'seafood'
}, {
  optionText: 'Vegetarian',
  optionValue: 'vegetarian'
}];

<RadioGroup
  name="plates"
  radioOptions={options}
  inline={false}
/>
```

Each radio is displayed as an error when the `hasError` prop is true. An errored radio group with a default checked option is rendered with a filled radio button.

```
const options = [{
  optionText: 'First',
  optionValue: '1'
}, {
  optionText: 'Second',
  optionValue: '2'
}, {
  optionText: 'Third',
  optionValue: '3'
}];

<RadioGroup
  name="placings"
  radioOptions={options}
  hasError
/>
```

```
const options = [{
  optionText: 'Planes',
  optionValue: 'plane'
}, {
  optionText: 'Trains',
  optionValue: 'train'
}, {
  optionText: 'Automobiles',
  optionValue: 'automobile'
}];

<RadioGroup
  name="transports"
  radioOptions={options}
  checkedValue="plane"
  hasError
/>
```

Each radio is disabled when the `disableAllOptions` prop is true. A disabled radio group with a default checked option is rendered with a filled radio button.

```
const options = [{
  optionText: 'Apple',
  optionValue: 'apple'
}, {
  optionText: 'Banana',
  optionValue: 'banana'
}, {
  optionText: 'Pear',
  optionValue: 'pear'
}];

<RadioGroup
  name="fruits"
  radioOptions={options}
  disableAllOptions
/>
```

```
const options = [{
  optionText: 'Books',
  optionValue: 'books'
}, {
  optionText: 'Television',
  optionValue: 'tv'
}];

<RadioGroup
  name="media-preference"
  radioOptions={options}
  disableAllOptions
  checkedValue="books"
/>
```

Individual options can also be disabled.
```
const options = [{
  optionText: 'Hamburger',
  optionValue: 'burger'
}, {
  optionText: 'Cheeseburger',
  optionValue: 'cheeseburger'
}, {
  optionText: 'Hotdog',
  optionValue: 'hotdog',
  disabled: true
}];

<RadioGroup
  name="cookout-food"
  radioOptions={options}
/>
```
