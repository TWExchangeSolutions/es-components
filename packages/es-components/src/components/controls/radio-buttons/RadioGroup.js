import React from 'react';
import PropTypes from 'prop-types';
import { noop } from 'lodash';
import viaTheme from 'es-components-via-theme';

import Fieldset from '../../containers/fieldset/Fieldset';

import RadioButton from './RadioButton';
import { ThemeProvider, withTheme } from 'styled-components';

export function RadioGroup({
  name,
  radioOptions,
  legendContent,
  value,
  hasError = false,
  disableAllOptions = false,
  inline = true,
  onClick = noop,
  theme,
  extraContent
}) {
  return (
    <ThemeProvider theme={theme}>
      <Fieldset legendContent={legendContent} extraContent={extraContent}>
        {radioOptions.map((config, index) => {
          const radioId = `${name}-option-${index + 1}`;
          const checked = config.optionValue === value;
          const isDisabled = disableAllOptions || config.isDisabled;
          const radioProps = {
            name,
            checked,
            hasError,
            isDisabled,
            inline,
            onClick,
            id: radioId,
            optionText: config.optionText,
            value: config.optionValue,
            ariaHide: extraContent !== undefined,
            theme
          };
          return <RadioButton key={radioId} {...radioProps} />;
        })}
      </Fieldset>
    </ThemeProvider>
  );
}

const radioOptionShape = {
  optionText: PropTypes.string.isRequired,
  optionValue: PropTypes.any.isRequired,
  /** Render this option as disabled */
  isDisabled: PropTypes.bool
};

RadioGroup.propTypes = {
  /** The name of the radio group */
  name: PropTypes.string.isRequired,
  /** The content to render in the legend */
  legendContent: PropTypes.node,
  /** Options for the radio group to display */
  radioOptions: PropTypes.arrayOf(PropTypes.shape(radioOptionShape)).isRequired,
  /** Selected option for the radio group */
  value: PropTypes.any,
  /** Display all radio buttons in an errored state */
  hasError: PropTypes.bool,
  /** Disable all radio buttons */
  disableAllOptions: PropTypes.bool,
  /** Display the radio buttons inline */
  inline: PropTypes.bool,
  /** Function to execute when a radio button is selected */
  onClick: PropTypes.func,
  /** Extra content that can be rendered after the Legend but before the radio buttons, allows
   * content to be put in that will not affect the accessability of the Legend/Radio button relationship.
   */
  extraContent: PropTypes.node,
  theme: PropTypes.object
};

RadioGroup.defaultProps = {
  theme: viaTheme
};

export default withTheme(RadioGroup);
