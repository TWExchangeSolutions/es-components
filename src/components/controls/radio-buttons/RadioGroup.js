import React from 'react';
import PropTypes from 'prop-types';
import { noop } from 'lodash';

import Fieldset from '../../base/containers/Fieldset';
import RadioButton from './RadioButton';

const radioOptionShape = {
  optionText: PropTypes.string.isRequired,
  optionValue: PropTypes.any.isRequired,
  /** Render this option as disabled */
  disabled: PropTypes.bool
};

class RadioGroup extends React.Component {
  static propTypes = {
    /** The name of the radio group */
    name: PropTypes.string.isRequired,
    /** The text to render in the legend */
    legendText: PropTypes.string,
    /** Options for the radio group to display */
    radioOptions: PropTypes.arrayOf(PropTypes.shape(radioOptionShape))
      .isRequired,
    /** Selected option for the radio group */
    checkedValue: PropTypes.any,
    /** Display all radio buttons in an errored state */
    hasError: PropTypes.bool,
    /** Disable all radio buttons */
    disableAllOptions: PropTypes.bool,
    /** Display the radio buttons inline */
    inline: PropTypes.bool,
    /** Function to execute when the value of the radio group changes */
    onCheckedValueChanged: PropTypes.func
  };

  static defaultProps = {
    hasError: false,
    disableAllOptions: false,
    inline: true,
    onCheckedValueChanged: noop
  };

  constructor(props) {
    super(props);

    this.handleOnChange = this.handleOnChange.bind(this);

    const { checkedValue } = this.props;

    this.state = {
      checkedValue
    };
  }

  handleOnChange(event) {
    const { value: checkedValue } = event.currentTarget;
    this.setState(
      {
        checkedValue
      },
      () => this.props.onCheckedValueChanged(checkedValue)
    );
  }

  render() {
    const {
      name,
      radioOptions,
      inline,
      legendText,
      hasError: isInErrorState,
      disableAllOptions
    } = this.props;
    const { checkedValue } = this.state;

    return (
      <Fieldset legendText={legendText}>
        {radioOptions.map((config, index) => {
          const radioId = `${name}-option-${index + 1}`;
          const checked = config.optionValue === checkedValue;
          const disabled = disableAllOptions || config.disabled;
          const radioProps = {
            name,
            checked,
            isInErrorState,
            disabled,
            inline,
            id: radioId,
            onChange: this.handleOnChange,
            optionText: config.optionText,
            value: config.optionValue
          };
          return <RadioButton key={radioId} {...radioProps} />;
        })}
      </Fieldset>
    );
  }
}

export default RadioGroup;
