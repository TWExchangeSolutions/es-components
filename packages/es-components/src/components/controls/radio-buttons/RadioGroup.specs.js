/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { range } from 'lodash';

import RadioButton from './RadioButton';
import { RadioGroup } from './RadioGroup';

function buildOptions(numberOfOptions, optionIndexToDisable) {
  return range(0, numberOfOptions).map(idx => ({
    optionText: `Option ${idx}`,
    optionValue: idx,
    isDisabled: idx === optionIndexToDisable
  }));
}

describe('RadioGroup component', () => {
  let instance;
  let defaultOptions;

  beforeEach(() => {
    defaultOptions = buildOptions(3);

    instance = shallow(
      <RadioGroup name="test" radioOptions={defaultOptions} />
    );
  });

  it('renders a RadioButton for each option', () => {
    expect(instance.find(RadioButton).length).toBe(3);
  });

  it('renders each RadioButton as disabled when disableAllOptions is true', () => {
    instance.setProps({ disableAllOptions: true });

    const allDisabled = instance
      .find(RadioButton)
      .everyWhere(x => x.prop('isDisabled'));

    expect(allDisabled).toBe(true);
  });

  it('renders a specific RadioButton as disabled when that option is set to disabled', () => {
    const radioOptions = buildOptions(3, 0);
    instance.setProps({ radioOptions });

    const firstRadio = instance.find(RadioButton).first();
    expect(firstRadio.prop('isDisabled')).toBe(true);

    const lastRadio = instance.find(RadioButton).last();
    expect(lastRadio.prop('isDisabled')).toBe(false);
  });

  it('renders each radio in an error state when hasError is true', () => {
    instance.setProps({ hasError: true });

    const allErrored = instance
      .find(RadioButton)
      .everyWhere(radio => radio.prop('hasError'));

    expect(allErrored).toBe(true);
  });

  it('renders as expected', () => {
    const tree = renderer.create(
      <RadioGroup name="test" radioOptions={defaultOptions} />
    );

    expect(tree).toMatchSnapshot();
  });

  it('renders as expected with legend text', () => {
    const tree = renderer.create(
      <RadioGroup
        name="test"
        radioOptions={defaultOptions}
        legendContent="Test legend"
      />
    );

    expect(tree).toMatchSnapshot();
  });
});
