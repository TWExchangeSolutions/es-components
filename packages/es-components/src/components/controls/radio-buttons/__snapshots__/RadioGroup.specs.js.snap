// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup component renders as expected 1`] = `
<fieldset
  className="sc-bZQynM dSZIHB"
>
  <div />
  <label
    checked={false}
    className="sc-bwzfXH kVIXAD sc-bdVaJa kHjvzh"
    disabled={false}
    htmlFor="test-option-1"
  >
    <input
      className="sc-htpNat jeuMMj"
      disabled={false}
      id="test-option-1"
      name="test"
      onClick={[Function]}
      type="radio"
      value={0}
    />
    <span
      className="radio-fill sc-ifAKCX fnveVG"
      fill="#fff"
    />
    <span
      aria-hidden={false}
      className="sc-bxivhb bRNnnP"
    >
      Option 0
    </span>
  </label>
  <label
    checked={false}
    className="sc-bwzfXH kVIXAD sc-bdVaJa kHjvzh"
    disabled={false}
    htmlFor="test-option-2"
  >
    <input
      className="sc-htpNat jeuMMj"
      disabled={false}
      id="test-option-2"
      name="test"
      onClick={[Function]}
      type="radio"
      value={1}
    />
    <span
      className="radio-fill sc-ifAKCX fnveVG"
      fill="#fff"
    />
    <span
      aria-hidden={false}
      className="sc-bxivhb bRNnnP"
    >
      Option 1
    </span>
  </label>
  <label
    checked={false}
    className="sc-bwzfXH kVIXAD sc-bdVaJa kHjvzh"
    disabled={false}
    htmlFor="test-option-3"
  >
    <input
      className="sc-htpNat jeuMMj"
      disabled={false}
      id="test-option-3"
      name="test"
      onClick={[Function]}
      type="radio"
      value={2}
    />
    <span
      className="radio-fill sc-ifAKCX fnveVG"
      fill="#fff"
    />
    <span
      aria-hidden={false}
      className="sc-bxivhb bRNnnP"
    >
      Option 2
    </span>
  </label>
</fieldset>
`;

exports[`RadioGroup component renders as expected with legend text 1`] = `
<fieldset
  className="sc-bZQynM dSZIHB"
>
  <div>
    <legend
      className="sc-EHOje bJBAcN"
    >
      Test legend
    </legend>
  </div>
  <label
    checked={false}
    className="sc-bwzfXH kVIXAD sc-bdVaJa kHjvzh"
    disabled={false}
    htmlFor="test-option-1"
  >
    <input
      className="sc-htpNat jeuMMj"
      disabled={false}
      id="test-option-1"
      name="test"
      onClick={[Function]}
      type="radio"
      value={0}
    />
    <span
      className="radio-fill sc-ifAKCX fnveVG"
      fill="#fff"
    />
    <span
      aria-hidden={false}
      className="sc-bxivhb bRNnnP"
    >
      Option 0
    </span>
  </label>
  <label
    checked={false}
    className="sc-bwzfXH kVIXAD sc-bdVaJa kHjvzh"
    disabled={false}
    htmlFor="test-option-2"
  >
    <input
      className="sc-htpNat jeuMMj"
      disabled={false}
      id="test-option-2"
      name="test"
      onClick={[Function]}
      type="radio"
      value={1}
    />
    <span
      className="radio-fill sc-ifAKCX fnveVG"
      fill="#fff"
    />
    <span
      aria-hidden={false}
      className="sc-bxivhb bRNnnP"
    >
      Option 1
    </span>
  </label>
  <label
    checked={false}
    className="sc-bwzfXH kVIXAD sc-bdVaJa kHjvzh"
    disabled={false}
    htmlFor="test-option-3"
  >
    <input
      className="sc-htpNat jeuMMj"
      disabled={false}
      id="test-option-3"
      name="test"
      onClick={[Function]}
      type="radio"
      value={2}
    />
    <span
      className="radio-fill sc-ifAKCX fnveVG"
      fill="#fff"
    />
    <span
      aria-hidden={false}
      className="sc-bxivhb bRNnnP"
    >
      Option 2
    </span>
  </label>
</fieldset>
`;
